### PR TITLE
[ACS-8745] handle row selection in process list

### PR DESCRIPTION
**Pull request by g-jaskowski**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
There's no handling of selection in process list.

**What is the new behaviour?**
Added output emitting current selection.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts [hunk 1]
```diff
@@ -275,6 +281,16 @@ export class ProcessInstanceListComponent extends DataTableSchema implements OnC
         this.rowClick.emit(this.currentInstanceId);
     }
 
+    onRowSelect(event: CustomEvent) {
+        this.selectedInstances = [...event.detail.selection];
+        this.rowsSelected.emit(this.selectedInstances);
+    }
+
+    onRowUnselect(event: CustomEvent) {
```
[the remark this answers is not in the document]
Comment (g-jaskowski) [reply lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts]: Right, unnecessary copy-paste from task list. I've removed duplication, thank you!

File: lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts [hunk 2]
```diff
@@ -275,6 +281,16 @@ export class ProcessInstanceListComponent extends DataTableSchema implements OnC
         this.rowClick.emit(this.currentInstanceId);
     }
 
+    onRowSelect(event: CustomEvent) {
+        this.selectedInstances = [...event.detail.selection];
```
[the remark this answers is not in the document]
Comment (g-jaskowski) [reply lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts]: Same case, unnecessary mirroring of task list. Applied suggestion, thanks!

File: lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts [hunk 3]
```diff
@@ -275,6 +280,10 @@ export class ProcessInstanceListComponent extends DataTableSchema implements OnC
         this.rowClick.emit(this.currentInstanceId);
     }
 
+    onRowCheckboxToggle(event: CustomEvent) {
```
Comment (nikita-web-ua) [inline lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts:283]: mb we should add unit tests for new functionality?

File: docs/process-services/components/process-list.component.md [hunk 4]
```diff
@@ -77,6 +77,7 @@ when the process list is empty:
 | rowClick | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<string>` | Emitted when a row in the process list is clicked. |
 | showRowContextMenu | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`DataCellEvent`](../../../lib/core/src/lib/datatable/components/data-cell.event.ts)`>` | Emitted before the context menu is displayed for a row. |
 | success | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`ProcessListModel`](../../../lib/process-services/src/lib/process-list/models/process-list.model.ts)`>` | Emitted when the list of process instances has been loaded successfully from the server. |
+| rowsSelected | [`EventEmitter`](https://angular.io/api/core/EventEmitter)`<`[`ObjectDataRow[]`](../../../lib/core/src/lib/datatable/data/object-datarow.model.ts)`>` | Emmitted when rows is selected or unselected. |
```
[the remark this answers is not in the document]
Comment (g-jaskowski) [reply docs/process-services/components/process-list.component.md]: Fixed, thanks!

File: lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts (hunk 3 above)
Comment (g-jaskowski) [reply lib/process-services/src/lib/process-list/components/process-list/process-list.component.ts:283]: Added, thank you!

**Review by swapnil-verma-gl — APPROVED** [review]

**Review by nikita-web-ua — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

**Review by AleksanderSklorz — APPROVED** [review]

**Pull request merged**